### DISCUSSION TITLE
Higher genus manifolds

### DISCUSSION THREAD

**Discussion by FedeClaudi** [JuliaManifolds/Manifolds.jl#532] [Ideas]
Hi all, 

Following a conversation on Slack with @mateuszbaran  and @kellertuer I wanted to open this space to start talking about ways to implement higher genus manifolds. 
I'm happy to do to the work, but some guidance from folks with more experience with `Manifolds.jl` would be very super helpful. 

@mateuszbaran on Slack you mentioned that the best way might be to construct a higher genus manifolds from charts, I tend to agree as there doesn't seem to be a nice parametrization for them. Could you point me in the direction of an example of how this approach is used to construct other manifolds with `Manifolds.jl`? Thank you

**Comment by kellertuer** [comment]
Hi,
In general the way would be to implement a smooth atlas, see https://juliamanifolds.github.io/Manifolds.jl/latest/features/atlases.html, which we only use for one example for now, the stereographic atlas on the Sphere, see https://juliamanifolds.github.io/Manifolds.jl/latest/manifolds/sphere.html#Manifolds.StereographicAtlas

**Reply by mateuszbaran** [reply]
Hi!

Yes, exactly -- `StereographicAtlas` is currently the best example we have. You should be able to compute geodesics using `ODEExponentialRetraction` (I can fix errors if there are any), although note that this isn't well optimized currently (`solve_exp_ode` needs to be improved).

You can calculate distances based on `NLSolveInverseRetraction` and `ODEExponentialRetraction` but I think that a gradient-based approach using sensitivity analysis of `solve_exp_ode` should be much faster. I more or less see how to do that but I haven't started putting it together yet.

**Reply by mateuszbaran** [reply]
BTW, here is a link I've posted on Slack that may be helpful for defining charts on higher genus manifolds: https://math.stackexchange.com/questions/3427382/how-to-construct-a-g-genus-torus-by-identifying-sides-of-a-4g-gon . I think mathematically what we need at this point is an atlas covering such a higher genus manifold with a metric tensor and Christoffel symbols and we could work from there in Manifolds.jl.

**Reply by FedeClaudi** [reply]
Allright, thanks both of you for the help. 
I'll have a look during this week for ways to construct an atlas for such manifolds. I might just start with the 2-torus and we can see if there's a way to generalize it to other manifolds later.

**Reply by mateuszbaran** [reply]
Cool, I'm going to sketch a code for calculating geodesics on a 1-torus in my proposed approach.

**Reply by kellertuer** [reply]
Cool, also an atlas for the Torus would be great, since having a few atlases would be great to be sure out interface idea works :)

**Reply by FedeClaudi** [reply]
that's fair. Perhaps that's a good place to start for me to familiarize myself with the interface with a simpler problem.
